presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -620,16 +624,48 @@ public void registerTableForView(Table tableReference)
         tablesForView.push(requireNonNull(tableReference, "table is null"));
     }
 
+    public void registerTableForMaterializedView(Table table)
+    {
+        requireNonNull(table, "table is null");
+        tablesForMV.put(table, tablesForMV.getOrDefault(table, 0) + 1);
+    }
+
     public void unregisterTableForView()
     {
         tablesForView.pop();
     }
 
+    public void unregisterTableForMaterializedView(Table table)
+    {
+        tablesForMV.remove(table);
+    }
+
+    public void registerTableForPredicateStitching(SchemaTableName table, Expression predicate)
+    {
+        tablesForPredicateStitching.put(table, predicate);
+    }
+
+    public void unregisterTableForPredicateStitching(SchemaTableName table)
+    {
+        tablesForPredicateStitching.remove(table);
+    }
+
     public boolean hasTableInView(Table tableReference)
     {
         return tablesForView.contains(tableReference);
     }
 
+    //TODO: Better name?
+    public Integer getTableCountForMaterializedView(Table tableReference)
+    {
+        return tablesForMV.getOrDefault(tableReference, 0);
+    }
```

Comment:
```java
public boolean hasTableInMaterializedView(Table tableReference)
{
    return tablesForMaterializedView.contains(tableReference);
}

Reply:
As we discussed, this wont work. I will refactor this tomorrow.